tests/image_normalize_test.py

```diff
@@ -0,0 +1,244 @@
+    )
+
+
+def test_normalize_negative_scale_factor_raises(
+    sample_images: SampleImageStacksWithLogs,
+    dark_current_image: DarkCurrentImage,
+) -> None:
+    from ess.imaging.normalize import SamplePixelThreshold
+
+    cleansed_sample_image = apply_threshold_to_sample_images(
+        cleanse_sample_images(sample_images, dark_current_image),
+        SamplePixelThreshold(sc.scalar(0.0, unit="counts")),
+    )
+
+    with pytest.raises(ValueError, match="Scale factor must be positive,"):
+        normalize_sample_images(
+            samples=cleansed_sample_image,
+            background=BackgroundImage(dark_current_image),
+            factor=ScaleFactor(sc.scalar(-1.0, unit="dimensionless")),
+        )
+
+
+def test_normalize_workflow(
+    sample_images: RawSampleImageStacks,
+    open_beam_images: OpenBeamImageStacks,
+    dark_current_images: DarkCurrentImageStacks,
+) -> None:
+    expected_normalized_sample_images = sc.DataArray(
+        data=sc.array(
+            dims=["time", "dim_1", "dim_2"],
+            values=[
+                [[1 / 3 * (5 / 3), 1 / 3 * (5 / 3)], [1 / 3 * (5 / 3), 0.0]],
+                [[3 / 3 * (5 / 3), 3 / 3 * (5 / 3)], [3 / 3 * (5 / 3), 0.0]],
+            ],
+            unit="counts",
+        ),
+        coords={
+            "time": sc.array(dims=["time"], values=[1, 2], unit="s"),
+        },
+    )
+
+    wf = YmirImageNormalizationWorkflow()
+    wf[SampleImageStacksWithLogs] = sample_images
+    wf[OpenBeamImageStacks] = open_beam_images
+    wf[DarkCurrentImageStacks] = dark_current_images
+    mean_ob_warning_msg = (
+        "Computing average open beam image assuming constant exposure time."
+    )
+    mean_dc_warning_msg = (
+        "Computing average dark current image assuming constant exposure time."
+    )
+    mean_sample_warning_msg = (
+        "Computing average sample pixel counts assuming constant exposure time."
+    )
+    bg_image_warning_msg = (
+        "Computing average background pixel counts assuming constant exposure time."
+    )
+    normalize_warning_msg = (
+        "Computing normalized sample image stack assuming constant exposure time."
+    )
+    with (
+        # Following warnings are expected to be raise
+        # until we use the correct exposure time in the data files
+        pytest.warns(expected_warning=UserWarning, match=mean_ob_warning_msg),
+        pytest.warns(expected_warning=UserWarning, match=mean_dc_warning_msg),
+        pytest.warns(expected_warning=UserWarning, match=mean_sample_warning_msg),
+        pytest.warns(expected_warning=UserWarning, match=bg_image_warning_msg),
+        pytest.warns(expected_warning=UserWarning, match=normalize_warning_msg),
+    ):
+        normalized = wf.compute(NormalizedSampleImages)
+        assert isinstance(normalized, sc.DataArray)
+        assert normalized.sizes['time'] == 2
+        assert normalized.unit == "dimensionless"
+        assert_allclose(normalized, expected_normalized_sample_images)
```

Comment:
Can you also add some tests on the functions that apply the thresholds on the sample and dark images?

Reply:
Done...!